File: FE-CRUDMascotas/src/app/interfaces/mascota.ts
import { Raza } from './Raza';
import { Propietario } from './propietario';
export interface Mascota {
  id: number;
  nombre: string;
  color: string;
  edad: number;
  peso: number;
  fechaCreacion?: Date;
  NombreUsuario: Propietario;
  raza: Raza;
}

File: BE-CRUDMascotas/Models/Mascota.cs
using System.Text.Json.Serialization;
using Microsoft.EntityFrameworkCore.Storage.ValueConversion.Internal;

namespace BE_CRUDMascotas.Models
{
    public class Mascota
    {
        public int Id { get; set; }
        public string Nombre { get; set; }
        public string Color { get; set; }
        public int Edad { get; set; }
        public float Peso { get; set; }
        public DateTime FechaCreacion { get; set; }
        public int UsuarioId { get; set; }        
        public Usuario? NombreUsuario { get; set; }
        public int RazaId { get; set; }      
        public Raza? raza { get; set; }
    }

}
